com.unity.netcode.gameobjects/Runtime/Components/NetworkTransform.cs

```diff
@@ -3858,59 +4078,165 @@ private void UpdateInterpolation()
+            // Get the tick latency (ticks ago) as time (in the past) to process state updates in the queue.
+            var tickLatencyAsTime = timeSystem.TimeTicksAgo(tickLatency).Time;
+
+#if COM_UNITY_MODULES_PHYSICS || COM_UNITY_MODULES_PHYSICS2D
+            // If using rigid body for motion, then we need to increment
+            // our tick latency based on the number of times FixedUpdate
+            // is executed.
+            if (m_UseRigidbodyForMotion)
+            {
+                tickLatencyAsTime += m_FixedTimeFrameDelta;
+                currentTime += m_FixedTimeFrameDelta;
+            }
+#endif
+
+            // Smooth dampening and extrapolation specific:
+            // We clamp between the tick rate frequency and the tick latency x tick rate frequency
+            var minDeltaTime = timeSystem.FixedDeltaTimeAsDouble;
+
+            // Maximum delta time is the maximum time we will lerp between values. If the time exceeds this due to extreme
+            // latency then the value's interpolation rate will be accelerated to reach the goal and continue interpolating
+            // the next state updates.
+            var maxDeltaTime = tickLatency * minDeltaTime;
 
             // Now only update the interpolators for the portions of the transform being synchronized
             if (SynchronizePosition)
             {
-                if (PositionInterpolationType == InterpolationTypes.Lerp)
+                if (PositionLerpSmoothing)
                 {
                     m_PositionInterpolator.MaximumInterpolationTime = PositionMaxInterpolationTime;
-                    m_PositionInterpolator.Update(cachedDeltaTime, tickLatencyAsTime, cachedServerTime);
+                }
+                m_PositionInterpolator.LerpSmoothEnabled = PositionLerpSmoothing;
+
+                // If either of these two position interpolation related values have changed, then reset the current state being interpolated.
+                if (m_PreviousPositionInterpolationType != PositionInterpolationType || m_PreviousPositionLerpSmoothing != PositionLerpSmoothing)
+                {
+                    m_PreviousPositionInterpolationType = PositionInterpolationType;
+                    m_PreviousPositionLerpSmoothing = PositionLerpSmoothing;
+                    m_PositionInterpolator.ResetCurrentState();
+                }
+
+                if (PositionInterpolationType == InterpolationTypes.Lerp)
+                {
+                    m_PositionInterpolator.Update(cachedDeltaTime, tickLatencyAsTime, currentTime);
                 }
                 else
                 {
-                    m_PositionInterpolator.Update(cachedDeltaTime, tickLatencyAsTime, minDeltaTime, maxDeltaTime);
+                    m_PositionInterpolator.Update(cachedDeltaTime, tickLatencyAsTime, minDeltaTime, maxDeltaTime,
+                        PositionInterpolationType == InterpolationTypes.LerpExtrapolateBlend);
```

Comment:
Here this parameter is true when `PositionInterpolationType == InterpolationTypes.LerpExtrapolateBlend` but where you're using this variable the code comments indicate that this parameter refers to `PositionInterpolationType == InterpolationTypes.Lerp`

Reply:
This bool parameter is used to determine if the update is to use LerpExtrapolateBlend or if not then it uses SmoothDampening. Does the parameter description need to be updated for better clarity?
`/// <param name="isLerpAndExtrapolate">Determines whether to use smooth dampening or extrapolation.</param>`